pendulum_control/src/pendulum_demo.cpp

```diff
@@ -0,0 +1,189 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/memory_strategies.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/strategies/message_pool_memory_strategy.hpp>
+
+#include <rttest/rttest.h>
+
+#include <pendulum_msgs/msg/joint_command.hpp>
+#include <pendulum_msgs/msg/joint_state.hpp>
+
+#include "pendulum_controller.hpp"
+#include "pendulum_motor.hpp"
+#include "rtt_executor.hpp"
+
+struct OutputThreadArgs
+{
+  std::shared_ptr<pendulum_control::RttExecutor> executor;
+  std::shared_ptr<pendulum_control::PendulumMotor> pendulum_motor;
+  std::shared_ptr<pendulum_control::PendulumController> pendulum_controller;
+
+  OutputThreadArgs(std::shared_ptr<pendulum_control::RttExecutor> exec,
+    std::shared_ptr<pendulum_control::PendulumMotor> motor,
+    std::shared_ptr<pendulum_control::PendulumController> controller)
+  : executor(exec), pendulum_motor(motor), pendulum_controller(controller) {}
+};
+
+void * live_output_thread(void * args)
+{
+  OutputThreadArgs * ptr_args = static_cast<OutputThreadArgs *>(args);
+  auto executor = ptr_args->executor;
+  auto pendulum_motor = ptr_args->pendulum_motor;
+  auto pendulum_controller = ptr_args->pendulum_controller;
+
+  // background task: don't want to interfere with performance
+  if (rttest_set_sched_priority(0, SCHED_IDLE) != 0) {
+    perror("Couldn't set priority of output thread to IDLE");
+  }
+
+  struct rttest_results stats;
+
+  while (executor->is_rttest_ready()) {
+    // Collect statistics
+    if (executor->is_running()) {
+      executor->get_rtt_results(stats);
+
+      printf("Commanded motor angle: %f\n", pendulum_controller->get_command());
+      printf("Actual motor angle: %f\n", pendulum_motor->get_position());
+
+      printf("Mean latency: %f ns\n", stats.mean_latency);
+      printf("Min latency: %d ns\n", stats.min_latency);
+      printf("Max latency: %d ns\n", stats.max_latency);
+
+      printf("Minor pagefaults during execution: %lu\n", stats.minor_pagefaults);
+      printf("Major pagefaults during execution: %lu\n\n", stats.major_pagefaults);
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(2));
+  }
+  return 0;
+}
+
+using namespace rclcpp::strategies::message_pool_memory_strategy;
+using rclcpp::memory_strategies::static_memory_strategy::StaticMemoryStrategy;
+
+int main(int argc, char * argv[])
+{
+  // Initialization phase
+
+  // TODO(jacquelinekay) runtime tuning of controller properties and physical characteristics
+  pendulum_control::PendulumProperties properties;
+  pendulum_control::PIDProperties pid;
+
+  auto pendulum_motor =
+    std::make_shared<pendulum_control::PendulumMotor>(std::chrono::nanoseconds(970000),
+      properties);
+  auto pendulum_controller = std::make_shared<pendulum_control::PendulumController>(std::chrono::nanoseconds(
+        960000), pid);
```

Comment:
Wrapped lines should use one indentation level (+2 spaces). Not sure if uncrustify is picking this up though.

Same below.